### PR TITLE
Feat/testpurchases orders routes

### PR DESCRIPTION
**Pull request by WildSteph**

### CODE REVIEW

File: routes/purchasesOrders.js [hunk 1]
```diff
@@ -6,23 +6,30 @@ router.get('/', (req, res) => {
...
+        res.status(500).json({ message: 'There are no purchasesorders...' })
       )
     }
     res.json(results);
   })
 });
 
 router.get('/:id', (req, res) => {
+  const { id } = req.params
+  if (isNaN(id)) {
```
Comment (rocambille) [inline routes/purchasesOrders.js]: NaN is a special number value: the string 'foo' is not a number, but it isn't a NaN either.

you may follow this pattern instead: https://github.com/rocambille/back-to-mvc/blob/91b743e6d6e999aba7ee9792c26ef8103ba21ac3/index.js#L16

File: routes/purchasesOrders.js [hunk 2]
```diff
@@ -31,17 +38,17 @@ router.get('/:id', (req, res) => {
 
 router.post('/', (req, res) => {
   const formData = req.body;
-  if (formData.title === null) {
+  if (formData.DVM_id === "" || formData.quantity === "") {
```
Comment (rocambille) [inline routes/purchasesOrders.js]: you test if the fields are empty, but not if they are present. that was the purpose of the "== null": you need both tests ;)

File: test/app.integration.spec.js [hunk 3]
```diff
@@ -154,4 +154,87 @@ describe('Test Activities:', () => {
...
+    });
+});
+
+
+it('POST / field missing from purchasesOrders', (done) => {
+  request(app)
+    .post('/purchasesOrders')
+    .send({
+      DVM_id: "",
+      quantity: ""
```
Comment (rocambille) [inline test/app.integration.spec.js:215]: the fields are not missing, since they are here (with empty strings)

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by WildSteph** [comment]
is good ?

File: test/app.integration.spec.js [hunk 4]
```diff
@@ -154,4 +154,104 @@ describe('Test Activities:', () => {
...
+      .then(response => {
+        done();
+      });
+  });
+});
+
+it('GET / Id purchasesOrders : bad ID', (done) => {
+  request(app)
+    .get('/purchasesOrders/noId')
+    .expect(400)
```
Comment (rocambille) [inline test/app.integration.spec.js]: this should be a 404 ;)

File: test/app.integration.spec.js [hunk 5]
```diff
@@ -154,4 +154,104 @@ describe('Test Activities:', () => {
...
+      done();
+    });
+});
+
+it('POST / field null from purchasesOrders', (done) => {
+  request(app)
+    .post('/purchasesOrders')
+    .send({
+      DVM_id: null,
+      quantity: null
```
Comment (rocambille) [inline test/app.integration.spec.js]: remove these lines: just send an empty object

File: routes/purchasesOrders.js (hunk 1 above)
Comment (rocambille) [reply routes/purchasesOrders.js]: you can remove this if now

**Review by rocambille — CHANGES_REQUESTED** [review]

**Review by WildSteph — COMMENTED** [review]
he and there? no

File: routes/purchasesOrders.js [hunk 6]
```diff
@@ -14,8 +14,8 @@ router.get('/', (req, res) => {
 });
 
 router.get('/:id', (req, res) => {
-  const { id } = req.params
-  if (isNaN(id)) {
+  const id = parseInt(req.params.id);
+  if (isNaN(id)){
```
Comment (WildSteph) [inline routes/purchasesOrders.js]: Testing github for comment

**Review by WildSteph — COMMENTED** [review]
he and there? no

**Pull request merged**